sdk/textanalytics/ai-text-analytics/src/analyzeBatchActionsResult.ts

```diff
@@ -47,3 +155,312 @@ export interface PagedAnalyzeBatchActionsResult
    */
   statistics?: TextDocumentBatchStatistics;
 }
+
+/**
+ * The type of different actions supported by the begin analyze batch actions operation.
+ * @internal
+ */
+type TextAnalyticsActionType =
+  | "RecognizeCategorizedEntities"
+  | "RecognizePiiEntities"
+  | "ExtractKeyPhrases";
+
+/**
+ * The type of an action error with the type of the action that erred and its
+ * index in the list of input actions.
+ * @internal
+ */
+interface TextAnalyticsActionError {
+  /**
+   * A code describing the kind of error produced
+   */
+  readonly code: ErrorCode;
+  /**
+   * A message from the service explaining the error
+   */
+  readonly message: string;
+  /**
+   * The type of the action that erred
+   */
+  readonly type: TextAnalyticsActionType;
+  /**
+   * The index of the action that erred in the list of input actions
+   */
+  readonly index: number;
```

Comment:
do we need `index` here? Since the result list will have all objects (successful result and error result). When we parse the actionErrorTarget reference, we can create the ***ActionResult type list first. So we maintain the same order as the input from user |output from service. Then insert the parsed error as error object to the ***ActionResult.

Reply:
That is one way to do it but I prefer my approach because I think it is more clear.